### PR TITLE
Make restructure-namespaces rule configurable

### PR DESCRIPTION
**Pull request by mitchlloyd**
Let users provide a list of namespaces for destructuring.

cc @mixonic, @bantic, @rwjblue

### CODE REVIEW

**Comment by rwjblue** [comment]
:+1:

File: lib/rules/destructure-namespaces.js [hunk 1]
```diff
@@ -20,4 +27,13 @@ var rule = function(context) {
   };
 };
 
+rule.schema = [
```
Comment (mixonic) [inline lib/rules/destructure-namespaces.js:30]: What does a `schema` on a rule do?

**Comment by mixonic** [comment]
:clap: great improvement @mitchlloyd

File: lib/rules/destructure-namespaces.js (hunk 1 above)
Comment (mitchlloyd) [inline lib/rules/destructure-namespaces.js:30]: Schema validates configuration passed to a rule. Unfortunately I was not able to find an example of testing config validation errors in the official ESLint rules that I looked at. The happy path is tested: you can configure the rule according to the tests. But I couldn't figure out a conventional path to expect errors and have a passing test.

Comment (mixonic) [inline lib/rules/destructure-namespaces.js:30]: Very cool, thanks for the explanation!

**Pull request merged**